src/bootstrap/src/core/build_steps/tool.rs

```diff
@@ -319,18 +319,20 @@ pub(crate) fn get_tool_rustc_compiler(
     builder: &Builder<'_>,
     target_compiler: Compiler,
 ) -> Compiler {
-    if builder.download_rustc() && target_compiler.stage == 1 {
+    if target_compiler.is_forced_compiler() {
+        return target_compiler;
+    }
+
+    if builder.download_rustc() && target_compiler.stage > 0 {
         // We already have the stage 1 compiler, we don't need to cut the stage.
```

Comment:
Question: should this say "we already have the stage **N** compiler" (emphasis mine)?

Reply:
Oops, good catch